eth/backend.go

```diff
@@ -122,18 +122,13 @@ func New(stack *node.Node, config *ethconfig.Config) (*Ethereum, error) {
 	if !config.SyncMode.IsValid() {
 		return nil, fmt.Errorf("invalid sync mode %d", config.SyncMode)
 	}
-	// (PIP-35): Only enforce the minimum gas price of 25 gwei for amoy
-	if config.Genesis != nil && config.Genesis.Config.ChainID.Cmp(params.AmoyChainConfig.ChainID) == 0 {
-		if config.Miner.GasPrice == nil || config.Miner.GasPrice.Cmp(big.NewInt(params.BorDefaultMinerGasPrice)) != 0 {
-			log.Warn("Sanitizing invalid miner gas price", "provided", config.Miner.GasPrice, "updated", params.BorDefaultMinerGasPrice)
-			config.Miner.GasPrice = new(big.Int).SetUint64(params.BorDefaultMinerGasPrice)
-		}
-	} else {
-		if config.Miner.GasPrice == nil || config.Miner.GasPrice.Cmp(common.Big0) <= 0 {
-			log.Warn("Sanitizing invalid miner gas price", "provided", config.Miner.GasPrice, "updated", ethconfig.Defaults.Miner.GasPrice)
-			config.Miner.GasPrice = new(big.Int).Set(ethconfig.Defaults.Miner.GasPrice)
-		}
+
+	// PIP-35: Enforce min gas price to 25 gwei
+	if config.Miner.GasPrice == nil || config.Miner.GasPrice.Cmp(big.NewInt(params.BorDefaultMinerGasPrice)) != 0 {
+		log.Warn("Sanitizing invalid miner gas price", "provided", config.Miner.GasPrice, "updated", ethconfig.Defaults.Miner.GasPrice)
+		config.Miner.GasPrice = ethconfig.Defaults.Miner.GasPrice
```

Comment:
`ethconfig.Defaults.Miner.GasPrice` is same as `params.BorDefaultMinerGasPrice`, right?

Why can we not just do
```
config.Miner.GasPrice = new(big.Int).SetUint64(params.BorDefaultMinerGasPrice)
```

Reply:
This is the pattern which is followed at different places. 

Basically we use the value from `params` to compare but while setting them use the one coming from default. In this case, as both of them are same, it should be fine.